lib/ansible/modules/windows/win_zip.ps1

```diff
@@ -0,0 +1,253 @@
+                if ((Get-Variable -Name archive -ErrorAction SilentlyContinue) -and ($null -ne $archive)) {
+                    $archive.Dispose()
+                }
+                Remove-BrokenZip $dest
+                Fail-Json -obj $result -message "Error creating zip file '$dest': $($_.Exception.Message)"
+
+            }
+            finally {
+                # Call dispose to get rid of the unmanaged object
+                if ((Get-Variable -Name archive -ErrorAction SilentlyContinue) -and ($null -ne $archive)) {
+                    $archive.Dispose()
+                }
+            }
+        }
+        "Undefined" { # This should never be hit as we check, but leave in just in case the early check is modified.
+            #Fail-Json -obj $result -message "Could not find any available compression utilities: $($_.Exception.Message)"
+            Fail-Json -obj $result -message "Could not find any available compression utilities: PowerShell 5.0 and .NET Framework 4.5 are minimally required.  Please install the Windows Management Framework 5.0 or higher."
+        }
+    }
+
+}
+
+# BEGIN: Determine which type of compression utils to use: PowerShell, .NET or COM (to follow dev standards)
+# Try PowerShell native first, then .NET 4.5
+Add-Type -ErrorAction SilentlyContinue -TypeDefinition @"
+    public enum CompressionUtils
+    {
+        PowerShell,
+        DotNet,
+        Undefined
+    }
+"@
+
+$try_dotnet = $false
+try {
+    # Determines if the Compress-Archive command is available (preferred use)
+    # if not will check if dotnet libs are available (.NET 4.5 or greater)
+    Get-Command -Name Compress-Archive | Out-Null
+    $utils_available = [CompressionUtils]::PowerShell
+}
+catch {
+    $try_dotnet = $true
+}
+
+if ($try_dotnet) {
+    try {
+        # determines if .net 4.5 is available, if this fails we can't
+        # fall back to the legacy COM Shell.Application to extract the zip because
+        # the options don't allow it to be suppressed in the UI
+        Add-Type -AssemblyName System.IO.Compression.FileSystem | Out-Null
+        Add-Type -AssemblyName System.IO.Compression | Out-Null
+        $utils_available = [CompressionUtils]::DotNet
+    }
+    catch {
+        # default to Undefined
+        $utils_available = [CompressionUtils]::Undefined
+    }
+
+}
+
+$result.utils_used = $utils_available.ToString()
+# END: Determine which type of compression utils to use: PowerShell or .NET or COM (to follow dev standards)
+
+if ($utils_available -eq [CompressionUtils]::Undefined) {
+    Fail-Json -obj $result -message "Could not find any available compression utilities: PowerShell 5.0 and .NET Framework 4.5 are minimally required.  Please install the Windows Management Framework 5.0 or higher."
+}
+
+# Validate that the extension of the destination zip file is 'zip' - only one supported right now
+$src_ext = (Split-Path $dest -Leaf).Split('.')[-1]
+if ($src_ext -ne "zip") {
```

Comment:
Another reason to use .NET, it doesn't care about the extension at all and can output to whatever the user wants.